homeassistant/components/glances/__init__.py

```diff
@@ -50,8 +61,31 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return unload_ok
 
 
-def get_api(hass: HomeAssistant, entry_data: dict[str, Any]) -> Glances:
+async def get_api(hass: HomeAssistant, entry_data: dict[str, Any]) -> Glances:
     """Return the api from glances_api."""
     entry_data.pop(CONF_NAME, None)
+    entry_data.pop(CONF_VERSION, None)
+
     httpx_client = get_async_client(hass, verify_ssl=entry_data[CONF_VERIFY_SSL])
-    return Glances(httpx_client=httpx_client, **entry_data)
+    for version in (3, 2):
+        api = Glances(httpx_client=httpx_client, version=version, **entry_data)
+        try:
+            await api.get_ha_sensor_data()
+            _LOGGER.debug("Connected to Glances API v%s", version)
+            if version == 2:
+                async_create_issue(
+                    hass,
+                    DOMAIN,
+                    "deprecated_version",
+                    is_fixable=False,
+                    severity=IssueSeverity.WARNING,
+                    translation_key="deprecated_version",
+                )
+            return api
+        except GlancesApiNoDataAvailable as err:
+            _LOGGER.debug("Failed to connect to Glances API v%s: %s", version, err)
+    raise UnknownError("Could not connect to Glances API")
```

Comment:
This is not an unknown error. Name it eg `ServerVersionMismatch` and say that there's no known server version available out of version 2 or 3.